env0/data_project.go

```diff
@@ -101,6 +102,11 @@ func getProjectById(id string, meta interface{}) (client.Project, diag.Diagnosti
 	apiClient := meta.(client.ApiClientInterface)
 	project, err := apiClient.Project(id)
 	if err != nil {
+		if frerr, ok := err.(*http.FailedResponseError); ok {
+			if frerr.NotFound() {
+				return client.Project{}, diag.Errorf("Could not find a project with id: %s", id)
```

Comment:
Please add a test case for this in `data_project_test.go`
Mock 404 and expect this error

Reply:
This has been added.